docs/developer/release-team.md

```diff
@@ -0,0 +1,80 @@
+- Maintain and improve user facing documentation about releases, release policy and calendar
+- Update the CAPI Netlify book certificates and DNS
+- Update the clusterctl Homebrew formula
+- Create and maintain the GitHub release milestone
+- Maintain and improve and release automation, tooling & related developer docs
+- Track tasks needed to add support for new Kubernetes versions in the next CAPI release
+
+
+## Team Roles
+
+- **Release Lead**: responsible for coordinating release activities, assembling the release team, taking ultimate accountability for all release tasks to be completed on time, and ensuring that a retrospective happens. The lead is also responsible for ensuring a successor is selected and trained for future release cycles.
+- **Communications/Docs/Release Notes Manager**: Responsible for communicating key dates to the community, improving release process documentation, and polishing release notes. Also responsible for ensuring the user-facing Netlify book and provider upgrade documentation are up to date.
+- **CI Signal/Bug Triage/Automation Manager**: Assumes the responsability of the quality gate for the release and makes sure blocking issues and bugs are triaged and dealt with in a timely fashion. Helps improve release automation and tools.
+- **Shadow**: Any Release Team member may select one or more mentees to shadow the release process in order to help fulfill future Release Team staffing requirements and continue to grow the CAPI community in general.
+
+## Team Selection
+
+To start, the release team will be assembled by the release team lead based on volunteers. A call for volunteers can be made through the usual communication channels (office hours, Slack, mailing list, etc.). In the future, we may consider introducing an application process similar to the Kubernetes release team application process. 
+
+### Selection Criteria
+
+When assembling a release team, the release team lead should look for volunteers who:
+
+- Can commit to the amount of time required across the release cycle
+- Are enthusiastic about being on the release team
+- Are members of the Cluster Lifecycle SIG
+- Have some prior experience with contributing to CAPI releases (such as shadowing a role for a prior release)
+- Have diverse company affiliations (i.e. not all from the same company)
+
+## Time Commitment
+
+As a member of the release team, you should expect to spend approximately 4-8 hours a week on release related activites for the duration of the term. 
+
+Before you volunteer to be part of a CAPI release team, please make certain that your employer is aware and supportive of your commitment to the release team.
+
+## Why should I volunteer?
+
+Volunteering to be part of a CAPI release team is a great way to contribute to the community and to the release process:
+
+- Get more familiar with the CAPI release process
+- Create lasting relationships with other members of the community
+- Contribute to the CAPI project health
```

Comment:
@fabriziopandini please let me know if you have other ideas :)

Reply:
That's a great start, thanks. 
We can add more in follow-up iterations (as well as a "convince your boss" section)